packages/ui/src/common/hooks/useQueryNodeTransactionStatus.ts

```diff
@@ -20,6 +26,16 @@ export function useQueryNodeTransactionStatus(blockHash?: Hash | string, shouldS
     info('Failed to subscribe to query node state')
   }, [queryNodeStateError])
 
+  useEffect(() => {
+    if (loading || queryNodeStateError) {
+      const timeout = setTimeout(() => {
+        setStatus('confirmed')
+      }, 10_000)
+
+      return () => clearTimeout(timeout)
+    }
+  }, [loading, queryNodeStateError])
```

Comment:
IMO `queryNodeStateError` shouldn't reset the timeout:
```suggestion
  useEffect(() => {
    if (loading) {
      const timeout = setTimeout(() => {
        setStatus('confirmed')
      }, 10_000)

      return () => clearTimeout(timeout)
    }
  }, [loading])
```